compiler/optimizer/LocalAnalysis.cpp

```diff
@@ -488,7 +488,7 @@ int TR_LocalAnalysisInfo::hasOldExpressionOnRhs(TR::Node *node, bool recalcConta
 #endif
    bool seenWriteBarrier = false;
    int32_t storeNumChildren = node->getNumChildren();
-
+   bool isStoreIndirect = node->getOpCode().isStoreIndirect();
```

Comment:
I'd like to see a comment here explaining why this value needs to be cached. We should explicitly state in the comment that the node will be recreated below and the question of whether the original store is indirect or not may change. We should also note the fact that the recreation is done for convenience of syntactic comparison.

This is non-trivial stuff happening here for future readers of the code, and we don't want them confused or folding away this variable should they not be as keen to the details here.

Reply:
Good point! I'll make the change.